internal/client.go

```diff
@@ -94,6 +94,17 @@ type (
 		//	- InternalServiceError
 		SignalWorkflow(ctx context.Context, workflowID string, runID string, signalName string, arg interface{}) error
 
+		// SignalWithStartWorkflow sends a signals to a workflow in execution, if workflow is not running or not found,
+		// it starts workflow then sends the signal in transaction.
+		// - workflowID, runID, signalName, signalArg are same as SignalWorkflow's parameters
+		// - options, workflow, workflowArgs are same as StartWorkflow's parameters
+		// The errors it can return:
+		//  - EntityNotExistsError, if domain does not exists
+		//  - BadRequestError
+		//	- InternalServiceError
```

Comment:
What's the semantic of the call when RunID is specified? For example previous workflow with this RunID has completed? Are we going to start a workflow ignoring RunID? It looks weird to me. I think we want to eliminate RunID parameter from this call as well as from the thrift API.

Reply:
Removed RunID. Agree having runID here might confuse user. User now only use workflowID to find latest running workflow to signal with start.